src/ga4gh/vrs/models.py

```diff
@@ -523,21 +539,31 @@ class Allele(_VariationBase):
         ..., description='An expression of the sequence state'
     )
 
-    def ga4gh_serialize_as_version(self, as_version):
+    def ga4gh_serialize_as_version(self, as_version: PrevVrsVersion):
         """This method will return a serialized string following the conventions for
-        Allele serialization as defined in the VRS version specified by 'as_version`."""
+        Allele serialization as defined in the VRS version specified by 'as_version`.
+
+        :raises ValueError: If ``state`` is not a ``LiteralSequenceExpression`` or
+            ``ReferenceLengthExpression``; ``state.sequence`` is null.
+        """
         location_digest = self.location.compute_digest(as_version=as_version)
+
+        if not isinstance(self.state, (LiteralSequenceExpression, ReferenceLengthExpression)):
+            err_msg = "Only `LiteralSequenceExpression` and `ReferenceLengthExpression` are supported for previous versions of VRS"
+            raise ValueError(err_msg)
+
         sequence = get_pydantic_root(self.state.sequence)
+
         if sequence is None:
```

Comment:
@ahwagner you had [made](https://github.com/ga4gh/vrs-python/pull/427#discussion_r1678545420) this comment. 

> LSE or RLE is okay here; both have a sequence field, and for small variants the RLE sequence is populated by default. However, I agree that we want to do some processing here as a separate PR; ideally we would implement the RLE to LSE function (it should work https://github.com/ga4gh/vrs/issues/426#issuecomment-1828387956) and just apply that in the case this is an RLE without a defined sequence.

However, for this if `sequence` is NOT provided then we'd have to make a call to the SeqRepo dataproxy and that felt weird to introduce. So I just decided to require `state.sequence` to be nonnull.

Reply:
That's a good call, I agree.